lib/parsers/gomod-graph-parser.ts

```diff
@@ -7,7 +7,11 @@ function parseGoModGraphLine(line: string): string[][] {
   return line
     .trim()
     .split(/\s/)
-    .map((item) => item.split('@'));
+    .map((item) => item.split('@'))
+    .map(([name, v]) => {
+      // Handle cases for prefixed-semver, see https://golang.org/ref/mod#major-version-suffixes
+      return [name.replace(/(.*)\/v[0-9](.*)/, '$1$2'), v];
```

Comment:
1. This doesn't support v10 (more than one number)
2. AFAIK versioning should be in the format of `example.com/mod/v2`, can you explain this regex?
3. You should probably extract the regex to a const and add the comments there

Reply:
4. What should happen for the following `example.com/mod/v2/v5`? at the moment it will return `example.com/mod/v2`

Reply:
1. Fixed
2. Go specifies modules can be of shape `modules/snyk/inner/v2` or `modules/snyk/v2/inner`
3. done

Reply:
4. `go mod graph` will fail for this